src/vs/editor/contrib/inlayHints/browser/inlayHintsContribution.ts

```diff
@@ -3,10 +3,10 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { registerEditorContribution } from 'vs/editor/browser/editorExtensions';
+import { EditorContributionInstantiation, registerEditorContribution } from 'vs/editor/browser/editorExtensions';
 import { HoverParticipantRegistry } from 'vs/editor/contrib/hover/browser/hoverTypes';
 import { InlayHintsController } from 'vs/editor/contrib/inlayHints/browser/inlayHintsController';
 import { InlayHintsHover } from 'vs/editor/contrib/inlayHints/browser/inlayHintsHover';
 
-registerEditorContribution(InlayHintsController.ID, InlayHintsController);
+registerEditorContribution(InlayHintsController.ID, InlayHintsController, EditorContributionInstantiation.Idle);
```

Comment:
Thanks for pointing that out. This is a bit surprising to me, I always thought of `IdleValue` and `runWhenIdle` as things which will eventually execute. If this is the case, I suggest we rename the class to `LazyValue` or something more clear.

I see that `runWhenIdle` allows to pass in a timeout, but `IdleValue` doesn't expose that. Rather than messing with `IdleValue` in this PR, I've added a change which does a loop over all editor contributions and accesses `.value` 5 seconds after editor startup, so the guarantee is that the contribution will be created at latest after 5s.

Reply:
> I always thought of IdleValue and runWhenIdle as things which will eventually execute. If this is the case, I suggest we rename the class to LazyValue or something more clear.

In observed reality they do eventually run but there is no guarantee. Naming is tricky, we actually already have [`Lazy`](https://github.com/microsoft/vscode/blob/26538eaaeff35324de3922a622dbfe51774abc8b/src/vs/base/common/lazy.ts#L20) but it is never running unless called. We advise to use it for things are very unlikely to ever be used (because it guarantees "slowness" on first access). The idle value is similar to lazy but can be lucky already be ready before first need. 

For editor contributions `runWhenIdle` might have been the better building block. The workbench is using that for its [eventually](https://github.com/microsoft/vscode/blob/86127c6cce7f27b71b8b112d0c59cc6ab6d5f4ad/src/vs/workbench/common/contributions.ts#L119) contributions: it creates as many contribution has possible in the remaining time and then reschedules, all with a 3sec timeout

> added a change which does a loop over all editor contributions and accesses .value 5 seconds after editor startup

Sounds like a fair way to counter regressions but I would still like code lenses and inlay hints to be eager. In worst-case they are appearing with a 5sec delay. Alternatively we can add another `EditorContributionInstantiation`-type like 'ModelOrSooner' which means when idle but latest when a model has been attached

Reply:
I've added multiple values for `EditorContributionInstantiation`